components/brave_ads/core/internal/account/confirmations/redeem_confirmation/redeem_confirmation_factory.cc

```diff
@@ -17,7 +18,7 @@ namespace brave_ads {
 void RedeemConfirmationFactory::BuildAndRedeemConfirmation(
     base::WeakPtr<RedeemConfirmationDelegate> delegate,
     const ConfirmationInfo& confirmation) {
-  if (!confirmation.opted_in) {
+  if (!UserHasOptedInToBravePrivateAds()) {
```

Comment:
now we can pass opted out confirmations to `RedeemOptedInConfirmation::CreateAndRedeem`